### PR TITLE
Remove power cells from light fixtures

### PR DESCRIPTION
**Pull request by hazamaitsuru**
:cl:
tweak: Light fixture power cells can now be removed instead of swapped
/:cl:

It didn't make sense to me that you couldn't remove the cell from a light fixture once it's been inserted, costing you the cell if you ended up deconstructing the fixture. You can now remove the cell with an empty hand; swapping cells has been removed as a consequence.

### CODE REVIEW

**Comment by hazamaitsuru** [comment]
Now you can remove a cell _with your mind!_

File: code/modules/power/lighting.dm [hunk 1]
```diff
@@ -84,6 +84,28 @@
...
+/obj/structure/light_construct/attack_hand(mob/user)
+	. = ..()
+	if(.)
+		return
+
+	if(cell)
+		user.visible_message("[user] removes \the [cell] from [src]!","<span class='notice'>You remove \the [cell].</span>")
+		user.put_in_hands(cell)
+		cell.update_icon()
+		src.cell = null
```
Comment (Jordie0608) [inline code/modules/power/lighting.dm]: Remove this `src.`

File: code/modules/power/lighting.dm [hunk 2]
```diff
@@ -84,6 +84,28 @@
...
+		add_fingerprint(user)
+		return
```
Comment (Jordie0608) [inline code/modules/power/lighting.dm]: And this return, they are unnecessary.

**Review by Jordie0608 — DISMISSED** [review]

File: code/modules/power/lighting.dm [hunk 3]
```diff
@@ -84,6 +84,28 @@
...
+
+/obj/structure/light_construct/attack_tk(mob/user)
+	if(!cell)
+		return
+
+	to_chat(user, "<span class='notice'>You telekinetically remove \the [cell].</span>")
+	cell.forceMove(loc)
```
Comment (Cyberboss) [inline code/modules/power/lighting.dm]:
```suggestion
	cell.forceMove(drop_location())
```

File: code/modules/power/lighting.dm [hunk 4]
```diff
@@ -84,6 +84,28 @@
...
+		cell.update_icon()
+		src.cell = null
+		add_fingerprint(user)
+		return
+
+/obj/structure/light_construct/attack_tk(mob/user)
+	if(!cell)
+		return
+
+	to_chat(user, "<span class='notice'>You telekinetically remove \the [cell].</span>")
```
Comment (Cyberboss) [inline code/modules/power/lighting.dm]:
```suggestion
	to_chat(user, "<span class='notice'>You telekinetically remove [cell].</span>")
```

File: code/modules/power/lighting.dm [hunk 5]
```diff
@@ -93,31 +115,33 @@
...
 		else
 			user.visible_message("<span class='notice'>[user] hooks up [W] to [src].</span>", \
 			"<span class='notice'>You add [W] to [src].</span>")
-		playsound(src, 'sound/machines/click.ogg', 50, TRUE)
-		W.forceMove(src)
-		cell = W
-		add_fingerprint(user)
-		return
+			playsound(src, 'sound/machines/click.ogg', 50, TRUE)
+			user.dropItemToGround(W)
```
Comment (Cyberboss) [inline code/modules/power/lighting.dm]:
```suggestion
			if (!user.temporarilyRemoveItemFromInventory(W))
						//add handling if the user can't drop the cell here
						return
```
You should probably move the visible message and sound below this too

File: code/modules/power/lighting.dm [hunk 6]
```diff
@@ -93,31 +115,33 @@
...
+				if(cell)
+					to_chat(user, "<span class='warning'>You have to remove the cell first!</span>")
+					return
+				else
+					to_chat(user, "<span class='notice'>You begin deconstructing [src]...</span>")
+					if (W.use_tool(src, user, 30, volume=50))
+						new /obj/item/stack/sheet/metal(drop_location(), sheets_refunded)
+						user.visible_message("[user.name] deconstructs [src].", \
+							"<span class='notice'>You deconstruct [src].</span>", "<span class='italics'>You hear a ratchet.</span>")
+						playsound(src.loc, 'sound/items/deconstruct.ogg', 75, 1)
```
Comment (Cyberboss) [inline code/modules/power/lighting.dm]:
```suggestion
						playsound(src, 'sound/items/deconstruct.ogg', 75, 1)
```

File: code/modules/power/lighting.dm [hunk 7]
```diff
@@ -84,6 +84,28 @@
 		to_chat(user, "<span class='danger'>This casing doesn't support power cells for backup power.</span>")
 		return
 
+/obj/structure/light_construct/attack_hand(mob/user)
+	. = ..()
+	if(.)
+		return
+
+	if(cell)
+		user.visible_message("[user] removes \the [cell] from [src]!","<span class='notice'>You remove \the [cell].</span>")
```
Comment (Cyberboss) [inline code/modules/power/lighting.dm]:
```suggestion
		user.visible_message("[user] removes [cell] from [src]!","<span class='notice'>You remove [cell].</span>")
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/power/lighting.dm (hunk 5 above)
Comment (hazamaitsuru) [reply code/modules/power/lighting.dm]: From what I could see, every other instance of this ifcheck simply returns, so I did that here as well.

**Comment by hazamaitsuru** [comment]
The requested changes have been performed.

File: code/modules/power/lighting.dm [hunk 8]
```diff
@@ -84,6 +84,27 @@
 		to_chat(user, "<span class='danger'>This casing doesn't support power cells for backup power.</span>")
 		return
 
+/obj/structure/light_construct/attack_hand(mob/user)
+	. = ..()
+	if(.)
+		return
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/power/lighting.dm]: what's the purpose of this

Comment (hazamaitsuru) [reply code/modules/power/lighting.dm]: I referenced the attack_hand function of something else. I read somewhere about nasty things and . = ..() so I kept it in.

File: code/modules/power/lighting.dm [hunk 9]
```diff
@@ -93,31 +114,34 @@
...
 			to_chat(user, "<span class='warning'>[W] is stuck to your hand!</span>")
 			return
-		user.dropItemToGround(W)
 		if(cell)
-			user.visible_message("<span class='notice'>[user] swaps [W] out for [src]'s cell.</span>", \
-			"<span class='notice'>You swap [src]'s power cells.</span>")
-			cell.forceMove(drop_location())
-			user.put_in_hands(cell)
+			to_chat(user, "<span class='warning'>There is a power cell already installed!</span>")
+			return
```
Comment (Cyberboss) [inline code/modules/power/lighting.dm]: Remove this return

File: code/modules/power/lighting.dm [hunk 10]
```diff
@@ -93,31 +114,34 @@
...
 		else
+			if(!user.temporarilyRemoveItemFromInventory(W))
```
Comment (Cyberboss) [inline code/modules/power/lighting.dm]: Condense this to
```dm
else if (user.temporarilyRemoveItemFromInventory(W))
```

File: code/modules/power/lighting.dm [hunk 11]
```diff
@@ -93,31 +114,34 @@
...
+				return
```
Comment (Cyberboss) [inline code/modules/power/lighting.dm]: Remove this return

File: code/modules/power/lighting.dm [hunk 12]
```diff
@@ -93,31 +114,34 @@
...
-		playsound(src, 'sound/machines/click.ogg', 50, TRUE)
-		W.forceMove(src)
-		cell = W
-		add_fingerprint(user)
-		return
+			playsound(src, 'sound/machines/click.ogg', 50, TRUE)
+			W.forceMove(src)
+			cell = W
+			add_fingerprint(user)
+			return
```
Comment (Cyberboss) [inline code/modules/power/lighting.dm]: And unindent this one

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by hazamaitsuru** [comment]
Changes 2: Electric Boogaloo

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**